beacon-chain/db/kv/blocks.go

```diff
@@ -204,6 +204,30 @@ func (s *Store) BlockRootsBySlot(ctx context.Context, slot types.Slot) (bool, []
 	return len(blockRoots) > 0, blockRoots, nil
 }
 
+// DeleteBlock from the db
+// This deletes the root entry from all buckets in the blocks DB
+// If the block is finalized this function panics
+func (s *Store) DeleteBlock(ctx context.Context, root [32]byte) error {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.DeleteBlock")
+	defer span.End()
+
+	return s.db.Update(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(finalizedBlockRootsIndexBucket)
+		if b := bkt.Get(root[:]); b != nil {
+			panic(errFinalizedInvalid)
```

Comment:
```suggestion
			return errFinalizedInvalid
```

Reply:
Do we need to delete the associated state as well? States are stored by block root I believe

Reply:
+1 

We shouldn't panic. Panic shouldn't be a DB behavior, rather a caller side behavior. Otherwise, we are assuming `DeleteBlock` could only be used for optimistic syncing

Reply:
Indeed, this was moved from my implementation in Kintsugi that was only for optimistic sync, I think it's fixed now, will panic in the caller but that's going to be in the Kintsugi branch